sql/sql_table.cc

```diff
@@ -9811,6 +9862,7 @@ static bool fk_prepare_copy_alter_table(THD *thd, TABLE *table,
                f_key->foreign_id->str, buff.c_ptr());
       DBUG_RETURN(true);
     }
+    case FK_COLUMN_NOT_NULL:
     default:
       DBUG_ASSERT(0);
```

Comment:
It would be good to add a source code comment that explains why this value is not possible here.